src/ccall.cpp

```diff
@@ -1676,6 +1676,20 @@ static jl_cgval_t emit_ccall(jl_value_t **args, size_t nargs, jl_codectx_t *ctx)
             emit_bitcast(ctx->ptlsStates, lrt),
             retboxed, rt, unionall, static_rt, ctx);
     }
+    if (fptr == (void(*)(void))&jl_threadid ||
+        ((!f_lib || (intptr_t)f_lib == 2) && f_name &&
```

Comment:
Agree, also for the 1 (and 0?).

Probably another PR though since it's used in many other places....